src/concrete/ml/common/preprocessors.py

```diff
@@ -829,29 +844,43 @@ def apply(self, graph: Graph) -> None:
             # 1. Optimize a and b for the subgraph
             # 2. Insert a and b in the graph
             # 3. Insert rounding to the graph
-            # 4. Insert a and b in the subgraph
-
+            # 4. Insert a and b in the subgraph of the TLU
+            #   We could also just modify the evalutor function to
+            #   include the scale-up.
+            #   That would require another way to do the optimization
+            #   since we rely on the subgraph to reduce the dimension
+            #   of the TLU.
+            
             # Get TLU sub-graph
-            if "subgraph" not in tlu_node.evaluator.properties["kwargs"]:
+            evaluator = tlu_node.evaluator
+            assert isinstance(evaluator, GenericEvaluator)
+
+            if "subgraph" not in evaluator.properties["kwargs"]:
+                # Not supported for now
```

Comment:
when does this case arise ? is it for fhe.univariate use?

Reply:
yes univariate or explicit table use in Concrete.

Anyway it's a good check since in the following we try to insert nodes in the subgraph